src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -284,40 +310,90 @@ export default class PhotoField extends React.Component {
   }
 
   onZoom = (e) => {
+    // Cropper returns the attempted zoom value
     const zoomValue = e.detail.ratio;
-    if (zoomValue < MAX_RATIO && zoomValue > MIN_RATIO) {
-      this.setState({ zoomValue });
-      this.updateBoundaryWarningAndButtonStates();
 
-      const moveBoundariesMet = getMoveBoundariesMet(this.refs.cropper.getCanvasData(), this.refs.cropper.getCropBoxData());
-      const zoomWarn = zoomValue >= WARN_RATIO;
+    // check if zoom is out of bounds
+    let zoomToBoundaryRatio;
+    if (zoomValue < this.state.minRatio) {
+      zoomToBoundaryRatio = this.state.minRatio;
+    } else if (zoomValue > this.state.maxRatio) {
+      zoomToBoundaryRatio = this.state.maxRatio;
+    }
 
-      this.setState({
-        warningMessage: makeCropBoundaryWarningMessage({ zoomWarn, ...moveBoundariesMet }),
-        ...makeMoveButtonsEnabledStates(moveBoundariesMet)
-      });
-    } else {
-      e.preventDefault();
+    // force zoom within zoom bounds
+    if (zoomToBoundaryRatio) {
+      this.refs.cropper.zoomTo(zoomToBoundaryRatio); // force zoom within constraints
+      e.preventDefault(); // prevents bad zoom attempt
+      return; // don't update state until the subsequent zoom attempt
     }
+
+    // zoom value is good- update the state / messaging
+    this.setState({ zoomValue });
+
+    const zoomWarn = zoomValue >= WARN_RATIO;
+    // at this point, the onZoom event has not resized the canvas-
+    // this forces the canvas back into move bounds if the zoom pulls a canvas edge into the cropBox
+    //   after the canvas has rendered with new width values
+    window.requestAnimationFrame(() => this.maybeMoveCanvasWithinBounds({}, zoomWarn));
   }
 
   setCropBox = () => {
-    const cropper = this.refs.cropper;
-    const slider = this.refs.slider;
-    const { width, naturalWidth } = this.refs.cropper.getImageData();
-    slider.value = width / naturalWidth;
-    const containerData = cropper.getContainerData();
-    const containerWidth = containerData.width;
-    const smallScreen = isSmallScreen(this.state.windowWidth);
-    const cropBoxSize = smallScreen ? 240 : 300;
-    const cropBoxLeftOffset = (containerWidth - cropBoxSize) / 2;
-    const cropBoxData = {
-      left: cropBoxLeftOffset,
-      top: 0,
-      width: cropBoxSize,
-      height: cropBoxSize
-    };
-    cropper.setCropBoxData(cropBoxData);
+    window.requestAnimationFrame(() => {
```

Comment:
@jbalboni This smoothed out the window resizing for me. I guess all of the events from `Cropper` happen before the canvas data is updated. Moving this logic into the next animation frame (i.e. when any pending layouts are complete) seems to smooth this out significantly. LMK if this improves the jankiness. I may do the same thing to other containerData dependent events.

Reply:
It definitely looks smoother, though I still see the issue with it leaving empty space in the crop box after resizing sometimes.

Reply:
Ah alright. I'm able to consistently recreate this. I think I know how to fix it.

- select image and start in small screen layout
- zoom into image a couple of times
- move image all of the way up
- make window larger